libvirt/resource_libvirt_domain_test.go

```diff
@@ -603,31 +603,45 @@ func TestAccLibvirtDomain_NetworkInterface(t *testing.T) {
 	resource "libvirt_domain" "%s" {
 		name              = "%s"
 		network_interface  {
-			network_name = "default"
+                        network_name = "${libvirt_network.%s.name}"
+                        hostname       = "myhost"
+			mac            = "52:54:00:A9:F5:17"
+			wait_for_lease = true
 		}
 		network_interface  {
-			network_name   = "default"
-			mac            = "52:54:00:A9:F5:17"
+                        network_id = "${libvirt_network.%s.id}"
+                        hostname       = "myhost"
+			mac            = "52:54:00:A9:F5:19"
 			wait_for_lease = true
 		}
 		disk {
 			file = "%s/testdata/tcl.iso"
 		}
-	}`, randomNetworkName, randomNetworkName, randomDomainName, randomDomainName, currentDir)
+	}`, randomNetworkName, randomNetworkName, randomDomainName, randomDomainName, randomNetworkName, randomNetworkName, currentDir)
 
 	resource.Test(t, resource.TestCase{
 		PreCheck:     func() { testAccPreCheck(t) },
 		Providers:    testAccProviders,
 		CheckDestroy: testAccCheckLibvirtDomainDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: config,
+				Config:             config,
+				ExpectNonEmptyPlan: false, // TODO: find why if set to true, the test fails
```

Comment:
we might need to check this before mergin because it could be a code smell of regression.

same also the `Destroy` I think we never used in Testacc

Reply:
I agree. I was unable to identify the reason, except that before the plan had the hostname but after the apply, it was empty. Terraform complained about this. I pointed this out in the issue https://github.com/dmacvicar/terraform-provider-libvirt/issues/708#issuecomment-592489392

Reply:
https://www.terraform.io/docs/extend/best-practices/testing.html#expecting-errors-or-non-empty-plans